bot/exts/moderation/modpings.py

```diff
@@ -116,16 +121,27 @@ async def add_role_schedule(self, mod: Member, work_time: int, schedule_start: d
             log.trace(f"Skipping adding moderator role to mod with ID {mod.id} - found in pings off cache.")
         else:
             log.trace(f"Applying moderator role to mod with ID {mod.id}")
-            await mod.add_roles(self.moderators_role, reason="Moderator scheduled time started!")
+            await self.reapply_role(mod, reason="Moderator scheduled time started!")
 
         log.trace(f"Sleeping for {work_time} seconds, worktime for mod with ID {mod.id}")
         await asyncio.sleep(work_time)
         await self.remove_role_schedule(mod, work_time, schedule_start)
 
-    async def reapply_role(self, mod: Member) -> None:
+    async def reapply_role(self, mod: Member, *, reason: str = None) -> None:
         """Reapply the moderator's role to the given moderator."""
+        await self.fetch_guild_task
+        if self.guild.get_role(Roles.mod_team) not in mod.roles:
+            log.warning(
```

Comment:
I don't think this needs to raise a sentry issue.

Either it can be just an info log, or it could send a message to Channels.mod_meta.

Reply:
Thinking on this more, we should remove them from the cache if they don't have the mod role too, otherwise this will just be hit on the next time the cog starts.

IMO this would make sense to be info log, remove from cache & message in mod-meta

Reply:
I think that is resolved with the rest of the enhancements in #2001.

Reply:
I can change this down to a info for now, and then in 2001 that can be changed.